src/System.IO.FileSystem/src/System/IO/Enumeration/MatchType.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.IO.Enumeration
+{
+    public enum MatchType
+    {
+        /// <summary>
+        /// Match using '*' and '?' wildcards.
+        /// </summary>
+        Simple,
+
+        /// <summary>
+        /// Match using DOS style matching semantics. '*', '?', '&lt;', '&gt;', and '"'
+        /// are all considered wildcards.
+        /// </summary>
+        Dos
```

Comment:
Modern shells (PowerShell, Bash) use more advanced globbing: '*' and '?' wildcards and _charsets like [a-z]._
PowerShell Core implementation works on Windows and Unix but expensive.
It will be great to have native implementation PowerShell-like globbing in .Net Core.

Also .Net Core always follow a platform behavior for file system case sensitivity. In PowerShell Core repo we have a feature request to implement case _insensitive_ IntelliSense on Unix - when a user types a file path/name it can help a good. It would be great to have a case sensitive parameter in file system APIs.

Reply:
@iSazonov Can you link to your globbing code? ASP.net has some algorithms too.

Reply:
@JeremyKuhne I see the PR merged - many thanks for the great work!

Is there an Issue for feedback about the new options and globbing?

[PowerShell globbing code](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/namespaces/LocationGlobber.cs) is possible too specific. I believe @SteveL-MSFT can give right contacts for comments.

Reply:
#18922 is probably the best place for globbing input/feedback. #25873 tracks the general extensiblity / performance work around enumeration.